pkg/detectors/zulipchat/zulipchat.go

```diff
@@ -12,18 +12,20 @@ import (
 	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
 )
 
-type Scanner struct{}
+type Scanner struct{
+	client *http.Client
+}
 
 // Ensure the Scanner satisfies the interface at compile time.
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	client = common.SaneHttpClient()
+	defaultClient = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat    = regexp.MustCompile(detectors.PrefixRegex([]string{"zulipchat"}) + common.BuildRegex(common.AlphaNumPattern, "", 32))
-	idPat     = regexp.MustCompile(detectors.PrefixRegex([]string{"zulipchat"}) + common.EmailPattern)
-	domainPat = regexp.MustCompile(detectors.PrefixRegex([]string{"zulipchat", "domain"}) + common.SubDomainPattern)
+	keyPat    = regexp.MustCompile(common.BuildRegex(common.AlphaNumPattern, "", 32))
```

Comment:
suggestion: For `keyPat` and `idPat` I think it's safe to leave the `detectors.PrefixRegex` since neither of them have the keyword or prefix directly within the regex. `domainPat` looks good since `zulipchat` is embedded within the regex itself.

Reply:
Due to prefixregex 40 char limit, i prefer to leave them out because then the zulipchat keyword needs to present in the line containing the idpat and keypat. If not present, detector wont work. 
Also , being 3 regex's in work and one containing zulipchat domain itself, i think it wont generate any false positives if we leave the prefix regex out as all 3 regexes need to match for this detector to work.